### ISSUE TITLE
Moving of folders not synced correctly [1.5.0beta]

### ISSUE THREAD

**Issue by jiri-matejka** [xbrowsersync/app#97]
### Steps to reproduce
1. Have folders with bigger amount of bookmarks (+-200)
2. Performs moves of these folders up the hierarchy, quickly
3. Do not wait for xBrowserSync to sync every move, and move multiple  folders

### Actual behaviour
I reported an error on one Firefox instance - stating that some changes were not propagated. Allright, I did the move for another time. It synced correctly it seemed.
But then I created new Firefox profile and the moves were not propagated there. The state was as before any moves happened.
Hopefully the bookmarks were not lost this time (which happened to me on xBrowserSync verson 1.4.0!!)

#### Log
I use official Firefox.
```
20190518165005398	[trace]	Installed v1.5.0beta6
20190518165005412	[trace]	Starting up
20190518165005436	[trace]	{"appVersion":"1.5.0beta6","platform":{"name":"firefox","version":"66.0.0","mobile":false,"os":"Windows 10"}}
20190518165015575	[trace]	Toolbar sync disabled
20190518165216668	[trace]	Sync 1gmn5rl (push) queued (1 waiting to sync)
20190518165220771	[trace]	Sync enabled
20190518165220882	[trace]	Sync 1gmn5rl completed (0 waiting to sync)
20190518165221000	[trace]	New sync id created: 6a879cda126340569590994f16bbf7cb
20190518170337856	[trace]	onMoved event detected
20190518170337876	[trace]	Sync qwmfzc (push) queued (1 waiting to sync)
20190518170340548	[trace]	onMoved event detected
20190518170340556	[trace]	Sync 8hyyi1 (push) queued (1 waiting to sync)
20190518170343201	[trace]	Sync qwmfzc completed (1 waiting to sync)
20190518170345093	[trace]	onMoved event detected
20190518170345098	[trace]	Sync 1e71788 (push) queued (1 waiting to sync)
20190518170347802	[trace]	Sync 8hyyi1 completed (1 waiting to sync)
20190518170351798	[trace]	Sync 1e71788 completed (0 waiting to sync)
20190518170354509	[trace]	onMoved event detected
20190518170354516	[trace]	Sync 17e73na (push) queued (1 waiting to sync)
20190518170356770	[trace]	onMoved event detected
20190518170356775	[trace]	Sync gdr45p (push) queued (1 waiting to sync)
20190518170357629	[error]	bookmarks.sync: [10111] XBookmarkNotFound
20190518170414686	[trace]	onRemoved event detected
20190518170414702	[trace]	Sync 1wklx5x (push) queued (1 waiting to sync)
20190518170415083	[error]	bookmarks.sync: [10114] DataOutOfSync
20190518170415099	[trace]	Sync 13j49h7 (pull) queued (1 waiting to sync)
20190518170415877	[trace]	Not clearing toolbar
20190518170421461	[trace]	Not populating toolbar
20190518170449214	[trace]	Waiting for sync: 13j49h7
20190518170454089	[trace]	Local population completed in 32.636s
20190518170454219	[trace]	Awaited sync complete: 13j49h7
20190518170454271	[trace]	Local sync data refreshed
20190518170513562	[trace]	onRemoved event detected
20190518170513581	[trace]	Sync jczuwp (push) queued (1 waiting to sync)
20190518170514584	[trace]	onRemoved event detected
20190518170514588	[trace]	Sync g9lc1b (push) queued (1 waiting to sync)
20190518170515718	[trace]	Sync jczuwp completed (1 waiting to sync)
20190518170517971	[trace]	Sync g9lc1b completed (0 waiting to sync)
```

**Comment by nero120** [comment]
Were you updating the sync from another browser at the same time? From the log it appears that a remote change was detected whilst syncing local updates.

**Comment by jiri-matejka** [comment]
Well as I described it was experiment in two different Firefox profiles on the same computer. This log is from instance A where I created totally new sync ID (as you can see in the log). 
I did the moves before signin in in my testing profile B so I am convinced there were no (real) remote updates (I trying to remember exactly so I could be mistaken). But I was very careful because xBrowserSync is fragile sometimes (the sync I mean, on the version 1.4.0 I even lost the bookmarks from the sync).
I could do the experiment again with sample set of bookmarks to help you.

**Comment by nero120** [comment]
Actually on further inspection of your log I can see that when you moved the bookmark at `20190518170356770` xbs was unable to find the moved local bookmark in the synced bookmarks data which then caused the proceeding out of sync error. I can see the previous moves were all succesful so it could be due to something about the bookmark that was moved, or some other factor.

Are you able to recreate the error by repeating the kind of moves you were doing at the time? If you can provide some repeatable steps so I can recreate it would help me greatly in resolving this annoying bug! 🙏

**Comment by nero120** [comment]
Please try again using the latest [beta](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta). I've included more logging and [raised a bug with Mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427) that affects moving or deleting multiple bookmarks simultaneously which could be a potential source of your issues.

If you do experience the issue again, please attach the xBrowserSync logs from all affected browsers.

**Comment by nero120** [comment]
@jiri-matejka would you mind trying the latest [v1.5.0beta11](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta)?

Please be aware however, due to a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427) in Firefox's bookmarks API, changing multiple bookmarks simultaneously will likely result in an error in xBrowserSync. The issues has been fixed in the latest [Firefox Nightly (v69.0a1)](https://www.mozilla.org/en-US/firefox/nightly/all/) so please download and test with Nightly so that you don't hit that known issue.

**Comment by nero120** [comment]
@jiri-matejka I've just released [v1.5.0](https://github.com/xbrowsersync/app/releases/tag/v1.5.0) which resolves most syncing issues, however there are [known issues](https://github.com/xbrowsersync/app/wiki/Known-issues) in Firefox due to bug's in Firefox's bookmarks API which can cause changes to be missed when performing many changes or simultaneous changes.

Unfortunately there's not much we can do until Mozilla fixes these issues. I am tracking the relevant bugs (e.g. 1556427](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427)) and will keep the pressure on Mozilla to resolve them asap.

I'm closing this issue but feel free to respond with any further comments.
